lib/kubeclient/common.rb

```diff
@@ -66,6 +71,15 @@ def initialize_client(
       end
     end
 
+    def method_missing(method_sym, *args, &block)
```

Comment:
@moolitayer I don't think the `super` part is needed. I have the feeling that this logic is simpler and may work as well:

```ruby
def method_missing(method_sym, *args, &block)
  discover unless !@discovered
  send(method_sym, *args, &block)
end
```